python/feathub/processors/flink/job_submitter/flink_session_cluster_job_submitter.py

```diff
@@ -52,49 +52,63 @@ def __init__(self, flink_processor: "FlinkProcessor"):
-                raise FeathubException(f"Features keys must not be None {features}.")
-            MemoryOnlineStore.get_instance().put(
-                table_name=sink.table_name,
-                features=table.to_pandas(),
-                schema=get_table_schema(features),
-                key_fields=features.keys,
-                timestamp_field=features.timestamp_field,
-                timestamp_format=features.timestamp_format,
-            )
-            return FlinkSessionClusterJob(None, self._executor)
-
-        with self.flink_processor.flink_table_builder.class_loader:
-            native_flink_table = self.flink_processor.flink_table_builder.build(
-                features=features,
-                keys=keys,
-                start_datetime=start_datetime,
-                end_datetime=end_datetime,
-            )
-
-            table_result = insert_into_sink(
-                self.flink_processor.flink_table_builder.t_env,
-                native_flink_table,
-                features,
-                sink,
-            )
-            return FlinkSessionClusterJob(table_result, self._executor)
+        flink_table_builder = self.flink_processor.flink_table_builder
+        with flink_table_builder.class_loader:
+            statement_set = None
+            for materialize_job in materialize_job_descriptors:
+                feature_descriptor = materialize_job.feature_descriptor
+                if not isinstance(feature_descriptor, TableDescriptor):
+                    raise FeathubException(
+                        f"The FeatureDescriptor is unresolved: {feature_descriptor}."
+                    )
+                table = FlinkTable(
+                    flink_processor=self.flink_processor,
+                    feature=feature_descriptor,
+                    keys=None,
+                    start_datetime=materialize_job.start_datetime,
+                    end_datetime=materialize_job.end_datetime,
+                )
+
+                if isinstance(materialize_job.sink, MemoryStoreSink):
+                    if feature_descriptor.keys is None:
+                        raise FeathubException(
+                            f"Features keys must not be None " f"{feature_descriptor}."
+                        )
+                    MemoryOnlineStore.get_instance().put(
+                        table_name=materialize_job.sink.table_name,
+                        features=table.to_pandas(),
+                        schema=get_table_schema(feature_descriptor),
+                        key_fields=feature_descriptor.keys,
+                        timestamp_field=feature_descriptor.timestamp_field,
+                        timestamp_format=feature_descriptor.timestamp_format,
+                    )
+                    continue
+
+                if statement_set is None:
+                    statement_set = flink_table_builder.t_env.create_statement_set()
+
+                native_flink_table = flink_table_builder.build(
```

Comment:
`flink_table_builder` clears tables that has been built after each invocation, which means the same feature descriptor might be resolved into multiple tables with multiple processor job. Can we optimize this part of logic and reuse the same table in all processor jobs?